### PR TITLE
Approaches: tweaks

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- approaches: use consistent general guidance headings
- approaches: use consistent "## Approach: " prefix

This is to prepare for extracting the approaches to individual approach documents.

### CODE REVIEW

File: exercises/practice/nucleotide-count/.approaches/introduction.md
Comment (cmcaine) [inline exercises/practice/nucleotide-count/.approaches/introduction.md:27]:
```suggestion
## Approach: using a variable for each nucleotide
```

I think this is the distinctive part of this approach, rather than the if/elseif

File: exercises/practice/pangram/.approaches/introduction.md
Comment (cmcaine) [inline exercises/practice/pangram/.approaches/introduction.md:28]:
```suggestion
This approach uses an anonymous function and uppercases the character rather than the string to avoid making a copy of the input.
```

File: exercises/practice/pythagorean-triplet/.approaches/introduction.md
Comment (cmcaine) [inline exercises/practice/pythagorean-triplet/.approaches/introduction.md:38]:
```suggestion
## Approach: quadratic time algorithm
```

File: exercises/practice/pythagorean-triplet/.approaches/introduction.md
Comment (cmcaine) [inline exercises/practice/pythagorean-triplet/.approaches/introduction.md:89]:
```suggestion
## Approach: linear time algorithm
```

File: exercises/practice/robot-name/.approaches/introduction.md
Comment (cmcaine) [inline exercises/practice/robot-name/.approaches/introduction.md:118]:
```suggestion
Just as 153 is 3 + 5 * 10 + 1 * 100,
```

The *s here are for multiplication, not emphasis. They don't need to be escaped if they're surrounded by whitespace.

File: exercises/practice/robot-name/.approaches/introduction.md
Comment (cmcaine) [inline exercises/practice/robot-name/.approaches/introduction.md:120]:
```suggestion
0 + 2 * 26 + 5 * 26 * 10
```

File: exercises/practice/rotational-cipher/.approaches/introduction.md
Comment (cmcaine) [inline exercises/practice/rotational-cipher/.approaches/introduction.md:8]: Almost any exercise with rotation will use modular arithmetic at some point, so I don't think this is a great name, but it's hard to think of a good one!

```suggestion
## Approach: using `Char` arithmetic and `map`
```

This is also generic, because most solutions will use `map`, but it's a bit more specific, imo.

File: exercises/practice/robot-name/.approaches/introduction.md
Comment (cmcaine) [inline exercises/practice/robot-name/.approaches/introduction.md:34]:
```suggestion
## Approach: encode names as numbers and shuffle them all in advance
```

Encoding names as numbers is the trickiest part of this, so let's lede (geddit) with that

File: exercises/practice/pythagorean-triplet/.approaches/introduction.md
Comment (cmcaine) [inline exercises/practice/pythagorean-triplet/.approaches/introduction.md:7]:
```suggestion
```

File: exercises/practice/pythagorean-triplet/.approaches/introduction.md
Comment (cmcaine) [inline exercises/practice/pythagorean-triplet/.approaches/introduction.md:6]:
```suggestion
```

File: exercises/practice/pythagorean-triplet/.approaches/introduction.md
Comment (cmcaine) [inline exercises/practice/pythagorean-triplet/.approaches/introduction.md:10]:
```suggestion
There are three reasonably common solutions to this problem, a cubic time solution, a quadratic time solution and a linear time solution
(if those terms are unclear to you, you might like to read about [time complexity](https://en.wikipedia.org/wiki/Time_complexity)).
A cubic time solution is easy to find, but too slow for the tests or for many practical uses.
A quadratic time solution is pretty easy to find and will be good enough for the tests.
Finding a linear time solution will require some mathematics. The solution below was derived by solving some simultaneous equations.
```

The three suggestions above combine the "General guidance" and "Approaches" sections.

File: exercises/practice/pangram/.approaches/introduction.md
Comment (cmcaine) [inline exercises/practice/pangram/.approaches/introduction.md:34]:
```suggestion
## Approach: marking bytes
```

Kinda bikesheddy, but I like that this and "counting bytes" have similar names because they're related approaches.

**Review by cmcaine — APPROVED** [review]
Thanks!

**Comment by cmcaine** [comment]
I'll add all my suggestions above and merge this evening if there are no comments :)

**Comment by ErikSchierboom** [comment]
Thanks for the review! I've applied all suggestions.

**Comment by cmcaine** [comment]
Thanks again for the interest and the PR :)

I'm still not sure that splitting them out will actually be better. Is it okay for this track to opt out of that for now and maybe Sascha and I can think about it?

**Comment by iHiD** [comment]
@cmcaine I'm happy for you personally to opt out, but not the track. We want to keep things consistent across the Exercism ecosystem as every time we don't it just makes life more confusing for students and means we have to build exceptions into our thinking which slows us down lots. So it's absolutely fine for you to choose not to split them out and keep adding things these wider documents, but if someone does split them out, we'd want to merge that PR 🙂 

Btw, if you want to read some of the discussion that went into this, a lot of it is documented at https://github.com/exercism/exercism/issues/6520 (lots of this conversation occurred on the weekly community calls, but I tried to put all the salient bits in that issue)

**Comment by cmcaine** [comment]
Well, I'm not going to pointlessly make work for people by doing it my own way if someone is then going to come along and reformat it.

I'm not going to tell you what to do, nor am I demanding that you explain yourself, but this is why I would prefer the Julia track opt out. Unless you have a good reason not to, I ask that you take it into consideration as you develop these feature and interact with maintainers/authors regarding it.

Splitting the approaches out into their own pages has costs as well as benefits. I have asked "why" a few times, and here are the benefits I have understood from your messages and links:

1. from Jeremy: that future tooling might be able to match student's solutions with approaches and do some clever stuff with that
2. from inspecting the C# dig-deeper pages myself: that it is easier for students to discover community videos if the approaches introduction page is shorter
3. guessing at future features from the discussion thread: if we ever have many many approaches, or user interaction on them, then a page per approach is better

Here are the main things I think a single page strategy does well, that a multipage strategy loses:

1. visibility of content: don't have to click a link to discover that there's a lot written about a topic & easier to scan or search for content
2. easier to write and especially easier to write a narrative if you know what content the reader has already been exposed to (they might not have read it, but they will have usually at least scrolled past it)
4. ease of use (no user interaction required beyond scrolling to access all information)

I feel like, at least on the julia track, the "representer" motivated strategy is unproven both in its means:

- when we prototyped a representer it was hard to categorise solutions well and it seemed harder for more complex exercises
- development of the future tooling is speculative

and its supposed gains: 

- no decision yet (as far as I can tell) on how the representer and approaches will be used
- no trials to test effectiveness of more advanced strategies versus simpler strategies that are well proven in other fields (text explaining expert solutions)

**Comment by cmcaine** [comment]
All that said, the bigger issue is about how exercism makes the dig deeper page visible to the student at all. When I finish an exercise it isn't advertised to me, it's just one more tab on quite a busy page.

**Comment by iHiD** [comment]
> I have asked "why" a few times, and here are the benefits I have understood from your messages and links:

Ah, I've not really answered why approaches are individual things. I only answered why approaches and articles are different. 

The main reason they're individual things is because it makes discovery much easier for students.

Take for example in the C# example, students can very quickly grok all the approaches and explore ones that are unfamiliar to them.

Screenshot 2022-11-17 at 22 17 06
Screenshot 2022-11-17 at 22 17 55
Screenshot 2022-11-17 at 22 17 00

With a long-post as the main format, we're asking them to scroll through pages of text to be able to quickly get an idea of the different approaches and detect what's interesting to them.

(Compare Julia's atbash cipher for example):
Screenshot 2022-11-17 at 22 17 38

That ability for a student to quickly decide where to put their attention is the most important reason for these being split out. In my experience, most people faced with a wall of text immediately just opt out, whereas when people have quickly scannable information, they're able to determine what they want and act on it.

> All that said, the bigger issue is about how exercism makes the dig deeper page visible to the student at all. When I finish an exercise it isn't advertised to me, it's just one more tab on quite a busy page.

Yep - this page needs work in general. We have various quick ideas for this, not least using the blue-dot paradigm we have within Exercism so that exercise is shown as needing you to look at something until it's read, but also want to reconsider this whole page layout in the future.

**Pull request merged**
